radiation/python/radsw/firstloop_gt4py.py

```diff
@@ -0,0 +1,428 @@
+            indict[var], backend, shape_2D, DTYPE_BOOL
+        )
+
+locdict_gt4py = dict()
+for var in locvars:
+    if var in ["tauae", "ssaae", "asyae", "taucw", "ssacw", "asycw"]:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, type_nbdsw)
+    elif var in ["zcf0", "zcf1"]:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_2D, DTYPE_FLT)
+    elif var in ["albbm", "albdf"]:
+        locdict_gt4py[var] = create_storage_zeros(
+            backend, shape_nlp1, (DTYPE_FLT, (2,))
+        )
+    elif var == "colamt":
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, type_maxgas)
+    else:
+        locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+for var in temvars:
+    locdict_gt4py[var] = create_storage_zeros(backend, shape_nlp1, DTYPE_FLT)
+
+s0 = 1368.22
+s0fac = indict["solcon"] / s0
+
+amdw = con_amd / con_amw
+amdo3 = con_amd / con_amo3
+
+
+@stencil(
+    backend=backend,
+    rebuild=rebuild,
+    verbose=True,
+    externals={
+        "solcon": indict["solcon"],
+        "s0fac": s0fac,
+        "con_g": con_g,
+        "con_avgd": con_avgd,
+        "con_amd": con_amd,
+        "con_amw": con_amw,
+        "amdw": amdw,
+        "amdo3": amdo3,
+        "iswrgas": iswrgas,
+        "iswcliq": iswcliq,
+        "iovrsw": iovrsw,
+        "nbdsw": nbdsw,
+        "ftiny": ftiny,
+        "oneminus": oneminus,
+    },
+)
+def firstloop(
+    plyr: FIELD_FLT,
+    plvl: FIELD_FLT,
+    tlyr: FIELD_FLT,
+    tlvl: FIELD_FLT,
+    qlyr: FIELD_FLT,
+    olyr: FIELD_FLT,
+    gasvmr: Field[type_10],
+    clouds: Field[type_9],
+    aerosols: Field[(DTYPE_FLT, (nbdsw, 3))],
+    sfcalb: Field[(DTYPE_FLT, (4,))],
+    dzlyr: FIELD_FLT,
+    delpin: FIELD_FLT,
+    de_lgth: FIELD_2D,
+    cosz: FIELD_2D,
+    idxday: FIELD_2DBOOL,
+    cosz1: FIELD_FLT,
+    sntz1: FIELD_FLT,
+    ssolar: FIELD_FLT,
+    albbm: Field[(DTYPE_FLT, (2,))],
+    albdf: Field[(DTYPE_FLT, (2,))],
+    tem1: FIELD_FLT,
+    tem2: FIELD_FLT,
+    pavel: FIELD_FLT,
+    tavel: FIELD_FLT,
+    h2ovmr: FIELD_FLT,
+    o3vmr: FIELD_FLT,
+    tem0: FIELD_FLT,
+    coldry: FIELD_FLT,
+    temcol: FIELD_FLT,
+    colamt: Field[type_maxgas],
+    colmol: FIELD_FLT,
+    tauae: Field[type_nbdsw],
+    ssaae: Field[type_nbdsw],
+    asyae: Field[type_nbdsw],
+    cfrac: FIELD_FLT,
+    cliqp: FIELD_FLT,
+    reliq: FIELD_FLT,
+    cicep: FIELD_FLT,
+    reice: FIELD_FLT,
+    cdat1: FIELD_FLT,
+    cdat2: FIELD_FLT,
+    cdat3: FIELD_FLT,
+    cdat4: FIELD_FLT,
+    zcf0: FIELD_2D,
+    zcf1: FIELD_2D,
+):
+    from __externals__ import (
+        solcon,
+        s0fac,
+        con_g,
+        con_avgd,
+        con_amd,
+        con_amw,
+        amdw,
+        amdo3,
+        iswrgas,
+        iswcliq,
+        iovrsw,
+        nbdsw,
+        ftiny,
+        oneminus,
+    )
+
+    with computation(FORWARD), interval(0, 1):
+        if idxday:
```

Comment:
I think it might help to include `if idxday[0,0]` here to make it extra obvious that it's a 2d field. but not required of course! Generally I prefer to not have [0,0,0] specified since it adds clutter, but in this class of stencil with higher and lower dimensional storages, it can help.

Reply:
I think I like it better without the [0, 0], since `idxday` is not actually used in any calculations, it doesn't seem super important to distinguish that it's a 2D field